### DISCUSSION TITLE
It is recommended to use the declare function package to ensure the normal operation of the plug-in under different versions

### DISCUSSION THREAD

**Discussion by qiyx2019** [babel/babel#14742] [Q&A] [answered]
import { declare } from "@babel/helper-plugin-utils";
export default declare((api, options, dirname) => {
return {};
});

**Comment by nicolo-ribaudo** [comment, marked as answer]
@qiyx2019 We use `declare` for two reasons:
- It provide "polyfills" for older Babel versions:
  - if your plugin needs to use `api.assertVersion`, it was not available in early Babel 7.0.0 beta prereleases
  - if your plugin needs to use `api.targets` or `api.assumptions`, they were not available in Babel 7.12.0
- If you are using TypeScript, you can install `@types/babel__helper-plugin-utils` to automatically get types for the `api` object.
